tests/functional/testSubscription.js

```diff
@@ -26,6 +27,114 @@ const SUBSCRIPTION_DETAILS = {
   },
 };
 
+describe('Subscription manager', function() {
+  describe('Expired subscription', function() {
+    this.timeout(30000);
+
+    // There are two constants defined below: userDataActive and
+    // userDataInactive which are used to override the Guardian endpoint to mock
+    // a subscription expiration by changing the VPN subscription from true to
+    // false. Because this override should be happening on the same device, the
+    // device is defined here and used later in both userDataActive and
+    // userDataInactive. Simply copying the same device definition in both cases
+    // will actually result in two distinct objects, causing the test to fail.
+    const device = {
+      name: 'Current device',
+      unique_id: '',
+      pubkey: '',
+      ipv4_address: '127.0.0.1',
+      ipv6_address: '::1',
+      created_at: new Date().toISOString()
+    };
+
+    const userDataActive = {
+      avatar: '',
+      display_name: 'Test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: true}},
+      devices: [device],
+    };
+    const userDataInactive = {
+      avatar: '',
+      display_name: 'Test test',
+      email: 'test@mozilla.com',
+      max_devices: 5,
+      subscriptions: {vpn: {active: false}},
+      devices: [device]
+    };
+
+    this.ctx.guardianOverrideEndpoints = {
+      GETs: {
+        '/api/v1/vpn/account': {
+          status: 200,
+          requiredHeaders: ['Authorization'],
+          body: userDataActive
+        }
+      },
+      POSTs: {
+        '/api/v1/vpn/device': {
+          status: 201,
+          requiredHeaders: ['Authorization'],
+          bodyValidator: validators.guardianDevice,
+          callback: (req) => {
+            device.name = req.body.name;
+            device.pubkey = req.body.pubkey;
+            device.unique_id = req.body.unique_id;
+          },
+          body: {}
+        },
+      }
+    };
+
+    it('Verify subscription before enabling VPN', async () => {
```

Comment:
```suggestion
    it('Prompts SubscriptionNeeded if a user's subscription is inactive on connection', async () => {
```
The test framework .describe( .it() ) are trying to nudge you to describe the expected behavior in the test name.